blockly/src/antlr/blocklyConditionVisitor.java

```diff
@@ -169,15 +185,14 @@ public INode visitRight_value(blocklyParser.Right_valueContext ctx) {
    */
   @Override
   public INode visitFunc_call(blocklyParser.Func_callContext ctx) {
-    String id = ctx.id.getText();
-    // Get boolean value
+    String id = ctx.id.getText(); // Get boolean value
     boolean boolVal =
         switch (id) {
-          case "naheWand" -> Server.isNearWall();
-          case "WandOben" -> Server.isNearWallUp();
-          case "WandUnten" -> Server.isNearWallDown();
-          case "WandLinks" -> Server.isNearWallLeft();
-          case "WandRechts" -> Server.isNearWallRight();
+          case "naheWand" -> httpServer.isNearWall();
```

Comment:
Hmmm. Eigentlich hätte ich hier sowas wie `Server.isNearWall()` erwartet. Da der Server ein Singleton ist, bräuchte man hier keine Instanz in einem Attribut zu halten.

Aber s.o.: Die Übergabe des Servers als Parameter erleichtert das Testen (bzw. macht es überhaupt erst möglich).

@Flamtky Wie siehst Du das?

Reply:
streiche die frage zu dependency injection vs. direktes nutzen des singletons. lass uns bei der dep. inj. bleiben.